lib/stdlib/src/uri_string.erl

```diff
@@ -375,6 +375,80 @@ recompose(Map) ->
     end.
 
 
+%%-------------------------------------------------------------------------
+%% Resolve URIs
+%%-------------------------------------------------------------------------
+-spec resolve(RefURI, BaseURI) -> TargetURI when
+      RefURI :: uri_string() | uri_map(),
+      BaseURI :: uri_string() | uri_map(),
+      TargetURI :: uri_string()
+                 | error().
+resolve(URIMap, BaseURIMap) ->
+    resolve(URIMap, BaseURIMap, []).
+
+
+-spec resolve(RefURI, BaseURI, Options) -> TargetURI when
+      RefURI :: uri_string() | uri_map(),
+      BaseURI :: uri_string() | uri_map(),
+      Options :: [return_map],
+      TargetURI :: uri_string() | uri_map()
+                 | error().
+resolve(URIMap, BaseURIMap, Options) ->
+    try resolve1(URIMap, BaseURIMap, Options)
+    catch
+        throw:{error, Atom, RestData} -> {error, Atom, RestData}
+    end.
+
+resolve1(URI, BaseURI, Options) ->
+	TargetURI = case resolve1(ensure_map_uri(URI), BaseURI) of
+		TargetURI0 = #{path := Path0} ->
+			%% We remove dot segments. Normalizing the entire URI
+			%% will sometimes add an extra slash we don't want.
+			#{path := Path} = normalize_map(#{path => Path0}),
+			TargetURI0#{path => Path};
+		TargetURI0 ->
+			TargetURI0
+	end,
+    case Options of
+        [return_map] ->
+            TargetURI;
+        [] ->
+            recompose(TargetURI)
+    end.
+
+resolve1(URI=#{scheme := _}, _) ->
+	URI;
+resolve1(URI=#{host := _}, BaseURI) ->
+	#{scheme := Scheme} = ensure_map_uri(BaseURI),
+	URI#{scheme => Scheme};
+resolve1(URI=#{path := <<>>}, BaseURI0) ->
+	BaseURI = ensure_map_uri(BaseURI0),
+	Keys = case maps:is_key(query, URI) of
+		true -> [scheme, host, port, path];
```

Comment:
+               true -> [scheme, host, port, userinfo, path];
Add also userinfo as "authority" has 3 subcomponents: host, port and userinfo.
Same applies to the other clauses.

Reply:
Done.